cpp/ql/src/experimental/Likely Bugs/ArrayAccessProductFlow.ql

```diff
@@ -34,7 +34,7 @@ class PhpEmalloc extends AllocationFunction {
   override int getSizeArg() { result = 0 }
 }
 
-predicate bounded(Instruction i, Bound b, int delta, boolean upper) {
+predicate bounded(Instruction i, SemBound b, int delta, boolean upper) {
```

Comment:
This looks like an actual change, not just moving stuff around?

Reply:
Oh, good point. `SemBound` is identical to `Bound`, so it's not an actual change. For some reason this query refers to internal IR-specific concepts rather than the `Sem` versions. For example, line 71 uses a `ValueNumberBound` instead of a regular `SemSsaBound`. That's why we need to import the `.internal.semantic.analysis.Bound` file instead of relying on `SemBound`.

Reply:
Line 71 says: `pai.getRight() = index and`. So I'm confused.

Reply:
Sorry. Line 70 🙂

Reply:
Ah, see it now. Thanks. It seems that this needs a slightly more substantial fix than just this one line change (l. 70 needs to be changed too), so maybe pull this out in a separate PR that does the proper fix?